### PR TITLE
feat: update event service and controller, tests

### PR DESCRIPTION
**Pull request by Sofiag8**
## Issue ticket number and link

https://trello.com/c/ot028BNc/350-migración-funcionalidad-editar-evento

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.

## Pull request type

Please check the type of change your PR introduces:

- [x] :sparkles: Feature
- [ ] :bug: Bugfix
- [ ] :pencil2: Fixing typos
- [ ] :lipstick: :art: Code style update (formatting, renaming)
- [ ] :recycle: Refactoring (no functional changes, no api changes)
- [ ] :zap: Performance
- [ ] :pencil: Documentation content changes
- [ ] :white_check_mark: Unit testing
- [ ] :ok_hand: Updating due to code review suggest
- [ ] :rocket: Release: v
- [ ] :wrench: Add or update configuration files
- [ ] :hammer: Add or update development scripts
- [ ] :bricks: Infrastructure related changes

## What is the current module?

We weren't using nestjs module for update events

## What is the new behavior?
now event module using nestjs has the update event functionality 

update an event passing the event id and receiving successful result with updated event

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/modules/events/dto/create-event.dto.ts [hunk 1]
```diff
@@ -37,6 +37,7 @@ export class CreateEventDto {
 }
 
 export enum EventStatus {
-  ACTIVE = 'ACTIVE',
-  INACTIVE = 'INACTIVE',
+  ACTIVE = 'active',
```
Comment (FelipeLosada1996) [inline src/modules/events/dto/create-event.dto.ts:40]: A good practice is that constant be in uppercase

File: src/modules/events/dto/response-update-event.dto.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { PartialType } from '@nestjs/mapped-types';
+import { CreateEventDto, EventStatus } from './create-event.dto';
+import { IsEnum, IsOptional } from 'class-validator';
+import { ResponseCreateDto } from './response-create-event.dto';
+
+export class ResponseUpdateDto extends ResponseCreateDto{}
```
Comment (FelipeLosada1996) [inline src/modules/events/dto/response-update-event.dto.ts:6]: For what is this?

Comment (Sofiag8) [reply src/modules/events/dto/response-update-event.dto.ts:6]: For now is the same that the created event response so is just an extension, in such case it change we can add fields there specifically for the update

File: src/modules/events/dto/create-event.dto.ts (hunk 1 above)
Comment (Sofiag8) [reply src/modules/events/dto/create-event.dto.ts:40]: Agree. For now we will handle it this way to do not change how it is working right now for v1 and how the front is handling statuses

File: src/modules/events/dto/response-update-event.dto.ts (hunk 2 above)
Comment (Sofiag8) [reply src/modules/events/dto/response-update-event.dto.ts:6]: is the Response dto when an event is updated @FelipeLosada1996

File: src/modules/events/event.controller.ts [hunk 3]
```diff
@@ -14,4 +15,11 @@ export class EventController {
 
     return await this.eventService.create(event)
   }
+
+  @Put('/:id')
+  @HttpCode(200)
+  async update(@Param('id') id: any, @Body() event: UpdateEventDto) {
```
Comment (FelipeLosada1996) [inline src/modules/events/event.controller.ts:21]: Define what type of data responds

File: src/modules/events/event.service.ts [hunk 4]
```diff
@@ -18,6 +20,12 @@ export class EventService {
     this.logger.log('Create event service')
     const event = EventMapper.createEventMapper(eventDto)
 
-    return await this.eventModel.create(event)
+    return this.eventModel.create(event)
+  }
+
+  async update(id: string, event: UpdateEventDto): Promise<ResponseUpdateDto | null> {
```
Comment (FelipeLosada1996) [inline src/modules/events/event.service.ts:26]: Why null? if the response if null, the ideal is launch an exception.
And the idea to return a data if with interface, a dto is using to input data.

File: src/modules/events/event.controller.ts (hunk 3 above)
Comment (Sofiag8) [reply src/modules/events/event.controller.ts]: Done

File: src/modules/events/event.service.ts (hunk 4 above)
Comment (Sofiag8) [reply src/modules/events/event.service.ts]: Fixed

File: src/modules/events/event.service.ts [hunk 5]
```diff
@@ -18,6 +20,14 @@ export class EventService {
...
 
-    return await this.eventModel.create(event)
+    return this.eventModel.create(event)
+  }
+
+  async update(id: string, event: UpdateEventDto): Promise<EventResponse> {
+    this.logger.log('Update event service')
+    const mappedEvent = EventMapper.updateEventMapper(event)
+    const updatedEvent = await this.eventModel.findOneAndUpdate({ _id: id}, mappedEvent, { new: true }).exec()
+    if (!updatedEvent) throw new NotFoundException()
```
Comment (FelipeLosada1996) [inline src/modules/events/event.service.ts:30]: Define a custom exception

**Review by kellyCSalazar — APPROVED** [review]

File: src/modules/events/event.service.ts [hunk 6]
```diff
@@ -21,7 +24,15 @@ export class EventService {
...
 
-    return await this.eventModel.create(event)
+    return this.eventModel.create(event)
+  }
+
+  async update(id: string, event: UpdateEventDto): Promise<EventResponse> {
+    this.logger.log('Update event service')
+    const mappedEvent = EventMapper.updateEventMapper(event)
+    const updatedEvent = await this.eventModel.findOneAndUpdate({ _id: id}, mappedEvent, { new: true }).exec()
+    if (!updatedEvent) throw new CustomNotFoundExceptionFilter(`Error at update event service, _id: ${id} was not found`,)
```
Comment (FelipeLosada1996) [inline src/modules/events/event.service.ts:34]: Something better could be ChupitosNotFoundException, reference to the app, no only custom

**Review by FelipeLosada1996 — APPROVED** [review]

**Review by kellyCSalazar — APPROVED** [review]

**Pull request merged**